chainer/training/extensions/log_report.py

```diff
@@ -56,17 +56,26 @@ class LogReport(extension.Extension):
             formatting. For example, users can use '{iteration}' to separate
             the log files for different iterations. If the log name is None, it
             does not output the log to any file.
+            It is recommended to suffix the filename with ``'.jsonl'`` when
+            ``file_format='jsonl'``.
             For historical reasons ``log_name`` is also accepted as an alias
             of this argument.
+        file_format (str): Format of the log file. If ``'jsonl'`` is specified,
+            a log file is in JSON Lines format, which is JSON objects delimited
+            by newlines (``LF``). It is suitable for stream processing and
+            works well with line-oriented tools in UNIX. If ``None``, it is a
+            single JSON list of the summaries.
 
     """
 
     def __init__(self, keys=None, trigger=(1, 'epoch'), postprocess=None,
-                 filename=None, **kwargs):
+                 filename=None, file_format=None, **kwargs):
```

Comment:
Let's make it a keyword-only argument. (see also [the policy](https://github.com/chainer/chainer/wiki/Development-policies/5acfc284fa9fecbea4998e95f7c127752476bc05#api))